mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala

```diff
@@ -23,71 +24,92 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
     toText(tpw).mkString(defaultWidth, false)
   }
 
-  def fullDefinition(sym: Symbol, tpe: Type): String = {
-    def name = nameString(sym)
-    keyString(sym) match {
-      case "" =>
-        val isImplicit = sym.is(Flags.Implicit)
-        val implicitKeyword = if (isImplicit) "implicit " else ""
-        s"$implicitKeyword$name: "
-      case key if sym.is(Flags.Method) =>
-        s"$key $name"
-      // package
-      case key if sym.is(Flags.Package) =>
-        val owners = for {
-          sym <- sym.ownersIterator
-          if !(sym.isRoot || sym.isEmptyPackage)
-          name = super.nameString(sym.name)
-        } yield name
-        "package " + owners.toList.reverse.mkString(".")
-      // enum case
-      case _ if sym.is(Flags.EnumVal) =>
-        s"case $name: "
-      // enum
-      case _ if sym.is(Flags.Enum) || sym.companionClass.is(Flags.Enum) =>
-        s"enum $name: "
-      // default
-      case key =>
-        // no need to add final on object, since they are all final
-        val finalKeyword =
-          if (sym.is(Flags.Final) && !sym.is(Flags.Module)) "final " else ""
-        s"$finalKeyword$key $name: "
-    }
-  }
-
   /**
    * for
    * - method: method signature
    *   - e.g. `[A: Ordering](x: List[Int]): A`
    * - otherwise: its shortened type
    *   - e.g. ` java.lang.String` ` Symbols.Symbol`
    */
-  def hoverDetails(
+  def hoverDetailString(
       sym: Symbol,
       history: ShortenedNames,
-      info: Type,
-      addFullDef: Boolean = true
+      info: Type
   )(using Context): String = {
-    val fullDef = if (addFullDef) fullDefinition(sym, info) else ""
-    // info is dealiased, while sym is not
     val typeSymbol = info.typeSymbol
+
+    def shortTypeString: String = {
+      val short = shortType(info, history)
+      s"${typeString(short)}"
+    }
+
+    def ownerTypeString: String =
+      typeSymbol.owner.fullName.stripModuleClassSuffix.toString
+
+    def name: String = nameString(sym)
+
+    def parentPackagesString: String = {
+      val owners = for {
+        sym <- sym.ownersIterator
+        if !(sym.isRoot || sym.isEmptyPackage)
+      } yield super.nameString(sym.name)
+      owners.toList.reverse.mkString(".")
```

Comment:
wdty about moving it into MtagsEnrichments?
There is a similar method: `fullNameBackticked` so the common part of `take to the root or empry` might be reused

Reply:
Actually, we can just use `fullNameBackticked` :sweat_smile: Removed the method altogether.